extensions/mssql/src/features.ts

```diff
@@ -921,3 +921,165 @@ export class SqlAssessmentServicesFeature extends SqlOpsFeature<undefined> {
 
 
 }
+
+export class ProfilerFeature extends SqlOpsFeature<undefined> {
```

Comment:
this is kinda the old way to do this.  Any reason not to use the ProfilerService.asFeature approach?   Also, I think you'll need to add an entry to the ClientOptions feature array (https://github.com/microsoft/azuredatastudio/blob/main/extensions/mssql/src/sqlToolsServer.ts#L152)

Reply:
I'll look at that alternative. Having started with basically a copy/paste from dataprotocolclient the next iteration should have a nice diff to show how the newer pattern works differently/better.

Reply:
sounds good, this approach also works.  I just think the alternate approach is bit better organized.

Reply:
is it something of a bug that sqlassessment seems to do both - it has an entry in features.ts and a separate service asFeature?

Reply:
I think there's a redundancy with a provider registration in src\sql\workbench\services\profiler\browser\interfaces.ts and src\sql\workbench\services\profiler\browser\profilerService.ts

I've tried launching my build using "Debug Renderer an Extension host processes" with a breakpoint set on 
ProfilerService.startSession in extensions\mssql\src\profiler\profilerService.ts, but that breakpoint is never hit.
Is that the right type of debug to use?

Should the profiler provider code in src\sql\workbench\api\browser\mainThreadDataProtocol.ts be removed or modified as well?

Reply:
this is the call stack of startSession in the UI 
![image](https://user-images.githubusercontent.com/2224906/91331192-74de6a80-e798-11ea-98ef-be31d8657725.png)
```
	public $registerProfilerProvider(providerId: string, handle: number): Promise<any> {
		const self = this;
		this._profilerService.registerProvider(providerId, <azdata.ProfilerProvider>{
			createSession(sessionId: string, createStatement: string, template: azdata.ProfilerSessionTemplate): Thenable<boolean> {
				return self._proxy.$createSession(handle, sessionId, createStatement, template);
			},
			startSession(sessionId: string, sessionName: string): Thenable<boolean> {
				return self._proxy.$startSession(handle, sessionId, sessionName);
			},
			stopSession(sessionId: string): Thenable<boolean> {
				return self._proxy.$stopSession(handle, sessionId);
			},
			pauseSession(sessionId: string): Thenable<boolean> {
				return self._proxy.$pauseSession(handle, sessionId);
			},
			getXEventSessions(sessionId: string): Thenable<string[]> {
				return self._proxy.$getXEventSessions(handle, sessionId);
			},
			connectSession(sessionId: string): Thenable<boolean> {
				return Promise.resolve(true);
			},
			disconnectSession(sessionId: string): Thenable<boolean> {
				return self._proxy.$disconnectSession(handle, sessionId);
			}
		});

		return undefined;
	}
```